stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -712,6 +717,8 @@ void onStart() {
       implements Executor {
     private static final Logger log = Logger.getLogger(ThreadlessExecutor.class.getName());
 
+    private static final Thread SHUTDOWN = new Thread(); // sentinel
```

Comment:
Creating a Thread worries me and seems like it is asking for trouble. Let's just make `waiter` an Object, which will either be null, a Thread, or a sentinel.